Update llm page (#818)
AaltoSciComp/scicomp-docs: aalto/llm-web-apis.rst

==================

> As a pilot service, :doc:`Aalto RSE </rse/index>` has a service

Reviewer: ```suggestion
As a pilot service, Aalto IT Services and :doc:`Aalto RSE </rse/index>` have a service
```

running some common open-source LLMS (Qwen3, gemma4, ...) available

> via the web.  This can be used for lightweight purposes via

Reviewer: ```suggestion
via the OpenAI-compatible APIs.  This can be used for lightweight purposes via
```

programming, but shouldn't replace batch usage (use

> :doc:`/triton/apps/llms`) or interactive chatting (use Aalto GPT).

Reviewer: ```suggestion
:doc:`/triton/apps/llms`) or interactive chatting (use the Aalto AI assistant).
```

Access
------

> The service is currently hosted on `https://llm-gateway.k8s.aalto.fi <https://llm-gateway.k8s.aalto.fi>`` and 

Reviewer: ```suggestion
The service is currently hosted on https://llm-gateway.k8s.aalto.fi and 
```

the API will remain there, while the front end will move in the near 
future. Access is currently limited to staff, and teachers can create
keys they can provide to students for teaching purposes. It is up 

> to the teacher to recall those keys again after the work is done.

Reviewer: Aalto networks only?

Reviewer: If we expect broad use including students, I think it would be useful if it's available even outside of aalto networks.

Author: I don't want to make this available outside aalto network. Students can (for all I know) go on VPN and use it. 
As to student use: 
The ultimate idea of this is that it can also serve the Azure endpoints at some point, and this will make it necessary to restrict usage/keys and/or associate them with some form of Cost center, so making it available to students directly is something I'm hesitant to do.

The API provided is basically a responses and chat/completions API 
and mostly OpenAI compatible. The service is based on vllm and the 

> api specification can be found `here. <https://llm-gateway.k8s.aalto.fi/docs>`

Reviewer: ```suggestion
api specification can be found `here <https://llm-gateway.k8s.aalto.fi/docs>`__.
```

> Here are some simple `examples. <https://github.com/AaltoSciComp/llm-examples/tree/main/aalto-llm-api>`__ 

Reviewer: ```suggestion
Here are some simple `examples <https://github.com/AaltoSciComp/llm-examples/tree/main/aalto-llm-api>`__.
```

Models
------

The cluster this service runs on has a limited amount of resources 
(there are only 8 GPUs in total in this cluster), and we hope that those we will
get more in the future. This means that most models are not running
constantly but scale down to 0 when not in use, and it can, depending on

> model size, take a substantial time for them to spin up.

Reviewer: ```suggestion
model size, take a substantial time (up to several minutes) for them to spin up (get the model loaded into memory).
```

On demand models are marked in the Model overview on the frontend. 

Request Additional Models
~~~~~~~~~~~~~~~~~~~~~~~~~
We can add additional models to the system if they are available on
huggingface and downloadable from europe. If you have a request for a model
please send an email to rse@aalto.fi with the name of the model and
we will try to set it up. 
Due to the resource limitations, the maximum model size we currently can

> support is 60GB ( which just about makes oss-gpt-120b possible). 

Reviewer: ```suggestion
support is 60GB (which just about makes oss-gpt-120b possible). 
```

Models run on local hardware on the Aalto University premises.
Kubernetes is used to manage computing power, so in principle there is

> plenty of opportunity for scaling and the models do scale to some extent.

Reviewer: ```suggestion
plenty of opportunity for increasing resources if there is high demand and the models do scale to some extent.
```
